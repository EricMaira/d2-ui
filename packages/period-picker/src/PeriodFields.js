import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Select from './Select';

const styles = theme => ({
    flexContainer: {
        display: 'flex',
        marginRight: -16,
    },
});

const PeriodFields = ({ periodFields, onChange, getValue, classes }) => (
    <div className={classes.flexContainer}>
        {Object.keys(periodFields).map(key => (
            <Select
                key={periodFields[key].name}
                name={periodFields[key].name}
                label={periodFields[key].label}
                value={getValue(periodFields[key].name)}
                onChange={onChange}
                options={periodFields[key].options}
            />
        ))}
    </div>
);

PeriodFields.propTypes = {
    periodFields: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    getValue: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(PeriodFields);
