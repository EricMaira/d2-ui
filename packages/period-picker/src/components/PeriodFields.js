import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import periodTypeLookup from '../periodTypes/lookup';
import Select from './Select';

const styles = theme => ({
    flexContainer: {
        display: 'flex',
        marginRight: -16,
    },
});

export const PeriodFields = ({ periodType, onChange, getValue, classes }) => {
    const periodFields = periodTypeLookup.get(periodType).getPeriodFields();

    return (
        <div className={classes.flexContainer}>
            {Object.keys(periodFields).map(key => (
                <Select
                    key={periodFields[key].name}
                    name={periodFields[key].name}
                    label={periodFields[key].label}
                    value={getValue(periodFields[key].name)}
                    onChange={onChange}
                    options={periodFields[key].options}
                />
            ))}
        </div>
    );
};

PeriodFields.propTypes = {
    periodType: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    getValue: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(PeriodFields);
