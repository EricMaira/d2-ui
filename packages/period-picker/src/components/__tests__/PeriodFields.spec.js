import React from 'react';
import { shallow } from 'enzyme';
import { PeriodFields } from '../PeriodFields';

describe('<PeriodFields/>', () => {
    it('Matches the snapshot', () => {
        const props = {
            periodType: 'Daily',
            onChange: jest.fn(),
            getValue: jest.fn(),
            classes: {
                flexContainer: 'flexContainer',
            },
        };
        const wrapper = shallow(<PeriodFields {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
