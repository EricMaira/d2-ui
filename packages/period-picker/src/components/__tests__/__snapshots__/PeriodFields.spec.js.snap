// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeriodFields/> Matches the snapshot 1`] = `
<div
  className="flexContainer"
>
  <WithStyles(Select)
    key="day"
    label="Day"
    name="day"
    onChange={[MockFunction]}
    options={
      Object {
        "1": "1",
        "10": "10",
        "11": "11",
        "12": "12",
        "13": "13",
        "14": "14",
        "15": "15",
        "16": "16",
        "17": "17",
        "18": "18",
        "19": "19",
        "2": "2",
        "20": "20",
        "21": "21",
        "22": "22",
        "23": "23",
        "24": "24",
        "25": "25",
        "26": "26",
        "27": "27",
        "28": "28",
        "29": "29",
        "3": "3",
        "30": "30",
        "31": "31",
        "4": "4",
        "5": "5",
        "6": "6",
        "7": "7",
        "8": "8",
        "9": "9",
      }
    }
  />
  <WithStyles(Select)
    key="month"
    label="Month"
    name="month"
    onChange={[MockFunction]}
    options={
      Object {
        "1": "jan",
        "10": "oct",
        "11": "nov",
        "12": "dec",
        "2": "feb",
        "3": "mar",
        "4": "apr",
        "5": "may",
        "6": "jun",
        "7": "jul",
        "8": "aug",
        "9": "sep",
      }
    }
  />
  <WithStyles(Select)
    key="year"
    label="Year"
    name="year"
    onChange={[MockFunction]}
    options={
      Object {
        "2014": "2014",
        "2015": "2015",
        "2016": "2016",
        "2017": "2017",
        "2018": "2018",
        "2019": "2019",
        "2020": "2020",
        "2021": "2021",
        "2022": "2022",
        "2023": "2023",
      }
    }
  />
</div>
`;
